src/vm/methodtablebuilder.cpp

```diff
@@ -2908,9 +2908,15 @@ MethodTableBuilder::EnumerateClassMethods()
             }
         }
 
-#ifndef FEATURE_DEFAULT_INTERFACES
         // Some interface checks.
-        if (fIsClassInterface)
+        // We only need them if default interface method support is disabled or if this is fragile crossgen
+#if !FEATURE_DEFAULT_INTERFACES || FEATURE_NATIVE_IMAGE_GENERATION
```

Comment:
Nit: CoreCLR uses "defined" for ifdefs everywhere, e.g.  `#if !defined(FEATURE_DEFAULT_INTERFACES) || defined(FEATURE_NATIVE_IMAGE_GENERATION)`. I am not sure why, but it would be nice to have a consistent style.